mp/importer/batcher.py

```diff
@@ -5,32 +5,45 @@
 The user can select the batch size, the maximum number of batches and the starting batch for yielding elements
 Other function will be added to randomly sample elements and to retrieve the last batch  
 """
-def run_in_batches(iterable, end_batch_callback=None, batch_size=2, batch_start=0, max_batches=None):	
+def run_in_batches(iterable, end_batch_callback=None, batch_size=2, batch_start=0, max_batches=None, seed=None, percentage=None):	
     """ Create batches from an iterable
    
     This function is a generator that creates batches of elements from an iterable 
     The callback function is called when a) maximum batches are reached or b) iterable is exhausted 
     All batches (complete and incomplete) are ended by a callback function
+
+    If random options are not None, gets a percentage of the total data
+    Pseudo-samples randomly the elements in the selected percentage (approximate)
+    The main script logs the seed to make it repeatable
     """ 	
+    import random
     c = 0
     ended = False
+    if percentage is not None:									
+        random.seed(seed)									# seed can be passed as kw to be repeatable
+        number = int(((100-percentage)*batch_size)/100)						# e.g. to yield the 30% of batch i mask the 70%
+        mask = set(random.sample(range(batch_size), number))										
+    else:
+        mask = []
```

Comment:
On Fri, Jun 05, 2015 at 02:00:33AM -0700, kiarasky wrote:
> >      c = 0
> >      ended = False
> > +    if percentage is not None:									
> > +        random.seed(seed)									# seed can be passed as kw to be repeatable
> > +        number = int(((100-percentage)*batch_size)/100)						# e.g. to yield the 30% of batch i mask the 70%
> > +        mask = set(random.sample(range(batch_size), number))										
> > +    else:
> > +        mask = []
> 
> ok, fixed. I'd like to make the tests work for py3, but i need to add a loop in the assert because the sampling methods give different results.         
>         if _PY2:
>              self.assertEqual(log_list, 'opqtvX')	  
>         if _PY3:
>             self.assertEqual(log_list, 'qrstuX')
> Also, I noticed that py3 is failing 2 tests, i'm not sure why but i'll check that

I think you want to pass the seed as an integer, then python 2 and 3
should do the same thing

-- 
Brian Sutherland